### DISCUSSION TITLE
ftp-server for tasmota/berry?

### DISCUSSION THREAD

**Discussion by waltkind** [arendst/Tasmota#21571] [Q&A] [answered]
In January there was good news that ftp-server is now merged for ESP8266 and ESP32.
There is also some text in the commands list, but I do not understand.

When I try the command UFSFTP 0, I get "Command":"Unknown".
Where do I need to insert the three definition lines?

**Comment by sfromis** [comment]
Doing a custom build with the settings you want is described on:
https://tasmota.github.io/docs/Compile-your-build/

In addition, there exists a sample FTP server coded in Berry, which you can copy to your device. This does not entail doing a build.
https://github.com/arendst/Tasmota/blob/development/tasmota/berry/drivers/ftp.be

**Reply by waltkind** [reply]
I already copied those three lines in my user_config_override.h and compiled this, but the answer to UFSFTP 0 was the same.
I will give your second hint a try.

**Reply by sfromis** [reply, marked as answer]
I just did a test build with those 3 settings, and it seems to work right away.
```
20:05:44.024 CMD: ufsftp
20:05:44.030 MQT: xxxxxxxx/stat/UFSFTP = {"UfsFTP":0}
```
Did you follow the instructions (at the end) for **where** to insert the lines in `user_config_override.h`?

**Reply by waltkind** [reply]
Thank you for your help.
I don't know what happened, but when I reflashed the binary that I compiled yesterday, it is working now.

A small word of critic to the documentation.  (I know it is hard to think for all stupid readers)
In https://tasmota.github.io/docs/Commands/#ftp-server there is not mentioned that you have to compile a new binary to use this commands. It is in the list of commands as all the other commands and only the already experienced user can assume from the `#define xxx` lines that this has to be added in the `user_config_override.h`

**Reply by sfromis** [reply]
That's why the docs is open for anyone to do edits, wiki-style (but with approval of changes)....
